src/components/commons/PrivatePage.js

```diff
@@ -0,0 +1,36 @@
+import { useContext, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import LoginContext from "../../contexts/LoginContext";
+import { setUser } from "../../services/localstorage";
+
+export default function PrivatePage({children}) {
+    const {config, setConfig, userData, setUserData} = useContext(LoginContext);
+
+    const data = JSON.parse(localStorage.getItem("linkr"));
+
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        if(!data) {
+            if(window.confirm("Please Login to continue")) {
+                navigate("/")
```

Comment:
**Suggestion:** it makes more sense that this file is inside a folder called decorator